### PR TITLE
in_splunk: Add documentation for input splunk plugin (Splunk HTTP HEC)

### PR DESCRIPTION
**Pull request by cosmo0920**
Corresponds to https://github.com/fluent/fluent-bit/pull/7349.

### CODE REVIEW

File: pipeline/inputs/splunk.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Splunk (HTTP HEC)
+
+The **splunk** input plugin handles Splunk HTTP HEC requests.
```
Comment (patrick-stephens) [inline pipeline/inputs/splunk.md:3]: Probably want to link out to the definition of what these Splunk acronyms are.

File: pipeline/inputs/splunk.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+| successful_response_code | It allows to set successful response code. `200`, `201` and `204` are supported.                                                              | 201     |
+| splunk\_token           | Add an Splunk token for HTTP HEC.`                                  |         |
+
+## Getting Started
+
+In order to start performing the checks, you can run the plugin from the command line or through the configuration file.
+
+#### How to set tag
+
+The tag for the Splunk input plugin is set by adding the tag to the end of the request URL by default.
```
Comment (patrick-stephens) [inline pipeline/inputs/splunk.md:23]: Can you show examples to clarify?

File: pipeline/inputs/splunk.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+This tag is then used to route the event through the system.
+The default behavior of the splunk input sets the tags for the following endpoints:
```
Comment (patrick-stephens) [inline pipeline/inputs/splunk.md:25]: I'm a little confused, it's probably a specific aspect of the Splunk protocol but what does this mean?

We receive an inbound request and the URL provided from that is one of these? We then process it?

File: pipeline/inputs/splunk.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+* `/services/collector`
+* `/services/collector/event`
+* `/services/collector/raw`
+
+The requests for these endpoints are interpreted as `services_collector`, `services_collector_event`, and `services_collector_raw`.
+
+If you want to use the other tags for multiple instantiating input splunk plugin, you have to specify `tag` property on the each of splunk plugin configurations to prevent collisions of data pipeline.
```
Comment (patrick-stephens) [inline pipeline/inputs/splunk.md:33]: Again, probably a bit confused and examples will help.

What happens if I set a tag without collision, does it override the defaults or is it ignored (that should be a warning I guess)?

File: pipeline/inputs/splunk.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+### Command Line
+
+From the command line you can configure Fluent Bit to handle Bulk API requests with the following options:
```
Comment (patrick-stephens) [inline pipeline/inputs/splunk.md:37]: Not sure why this is here, it doesn't look like we do anything to handle bulk requests. Is this a typo and it should be about just handling Splunk input or is there a differentiation between non-bulk and bulk - how do we enable either in that case?

File: pipeline/inputs/splunk.md (hunk 3 above)
Comment (cosmo0920) [reply pipeline/inputs/splunk.md:25]: The latter lines show the default behavior for handling when not specifying tag property in splunk input plugin.

**Pull request merged**